lib/logstash/outputs/kafka.rb

```diff
@@ -224,7 +224,7 @@ def multi_receive(events)
     end
 
     events.each do |event|
-      break if event == LogStash::SHUTDOWN
+      break if LOGSTASH_VERSION < '8.0' and event == LogStash::SHUTDOWN
```

Comment:
I'm not sure we need this line of code at all - AFAICT, the `LogStash::SHUTDOWN` event being passed to down to plugins is long removed from Logstash code (https://github.com/elastic/logstash/issues/8847#issuecomment-352573320 refers to it being removed in Logstash `5.0`). WDYT @jsvd?

However... we should also check how prevalent this check is within our plugins - I found one [other](https://github.com/logstash-plugins/logstash-output-nagios_nsca/blob/c81f81ea8decfdee73f5c7cd99454b51d2ad4ccd/lib/logstash/outputs/nagios_nsca.rb#L69) case where this check was made.

Reply:
I checked in all repositories inside `logstash-plugins` organization and for tier 1 and 2 it is used only in the `integration-kafka` and `output-kafka` (which is now deprecated) the rest of references are in other plugins, here is the summary:
```
Tier 1
logstash-integration-kafka/lib/logstash/outputs/kafka.rb:227
logstash-output-kafka/lib/logstash/outputs/kafka.rb:218:       


logstash-filter-collate/lib/logstash/filters/collate.rb
logstash-output-cloudwatch/lib/logstash/outputs/cloudwatch.rb:177:    
logstash-output-jira/lib/logstash/outputs/jira.rb:125:     
logstash-output-lumberjack/lib/logstash/outputs/lumberjack.rb:58:     
logstash-output-nagios_nsca/lib/logstash/outputs/nagios_nsca.rb:69:     
logstash-output-newrelic/lib/logstash/outputs/newrelic.rb:130:    
logstash-output-redmine/lib/logstash/outputs/redmine.rb:115:     
logstash-output-zookeeper/lib/logstash/outputs/zookeeper.rb:49:    
```

given the comment @robbavey  reported, I think we could completely remove the check.

Reply:
If we remove it we are backward and forward compatible

Reply:
+++ on removing this check, of the tier 2 I'd only prioritize lumberjack and cloudwatch outputs, as they're included in the logstash distribution

Reply:
Removed, PRs for the other two plugins:
- lumberjack output https://github.com/logstash-plugins/logstash-output-lumberjack/pull/31
- cloudwatch output https://github.com/logstash-plugins/logstash-output-cloudwatch/pull/18